packages/slice-machine/src/modules/slices/index.ts

```diff
@@ -164,85 +167,45 @@ export const slicesReducer: Reducer<SlicesStoreType | null, SlicesActions> = (
       return { ...state, libraries: newLibraries };
     }
     case getType(pushSliceCreator.success): {
-      const newComponentUI = action.payload.component;
-      const newRemoteSlices = [...state.remoteSlices];
+      const { component, updatedScreenshotsUrls } = action.payload;
 
-      const remoteSliceIndex = state.remoteSlices.findIndex(
-        ({ id }) => id === newComponentUI.model.id
+      const remoteSlice = state.remoteSlices.find(
+        (slice) => slice.id === component.model.id
       );
 
-      if (remoteSliceIndex !== -1) {
-        newRemoteSlices[remoteSliceIndex] = newComponentUI.model;
-      } else {
-        newRemoteSlices.push(newComponentUI.model);
-      }
-
-      const newLibraries = state.libraries.map((library) => {
-        if (library.name !== newComponentUI.from) return library;
+      const updateScreenshots = (remoteSlice: SliceSM): SliceSM => {
         return {
-          ...library,
-          components: library.components.map((component) => {
-            return component.model.id !== newComponentUI.model.id
-              ? component
-              : newComponentUI;
-          }),
+          ...remoteSlice,
+          variations: remoteSlice.variations.map((variation) => ({
+            ...variation,
+            imageUrl: updatedScreenshotsUrls[variation.id] || undefined,
+          })),
         };
-      });
+      };
+
+      const updatedRemoteSlices = remoteSlice
+        ? state.remoteSlices.map((remoteSlice) => {
+            if (remoteSlice.id !== component.model.id) return remoteSlice;
+            return updateScreenshots(remoteSlice);
+          })
+        : [...state.remoteSlices, updateScreenshots(component.model)];
 
       return {
         ...state,
-        libraries: newLibraries,
-        remoteSlices: newRemoteSlices,
+        remoteSlices: [...updatedRemoteSlices],
       };
     }
-    case getType(generateSliceScreenshotCreator.success): {
-      const { screenshots: screenshotUrls, component } = action.payload;
-
-      const newLibraries = state.libraries.map((library) => {
-        if (library.name !== component.from) return library;
-        return {
-          ...library,
-          components: library.components.map((c) => {
-            return component.model.id !== c.model.id
-              ? c
-              : { ...c, screenshotUrls };
-          }),
-        };
-      });
-
-      return { ...state, libraries: newLibraries };
-    }
+    case getType(generateSliceScreenshotCreator.success):
```

Comment:
two cases next to each other works as a fall-through.
so if `getType(generateSliceScreenshotCreator.success)` is true it will run the code block in `case getType(generateSliceCustomScreenshotCreator.success):`

is this what we want?

Reply:
yeah, they do nothing but update return the new component now